runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3178,15 +3178,47 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 {
 	MM_GCExtensions* extensions = MM_GCExtensions::getExtensions(vmThread);
 	J9JavaVM *vm = vmThread->javaVM;
-	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * and we don't change the original heap geometry from snapshot run.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	/* We re-use the softMx logic here to adjust default maximum heap size in restore path
+	 * temporarily.
+	 */
+	if (!extensions->userSpecifiedParameters._Xmx._wasSpecified){
+		uintptr_t candidateSoftMx = 0;
+		/* if the user have set maxRAMPercent through -XX:MaxRAMPercentage */
+		if (0.0 <= extensions->maxRAMPercent) {
+			candidateSoftMx = extensions->maxRAMPercent * extensions->usablePhysicalMemory / 100.0;
+		} else {
+			/* We use false here for computeDefaultMaxHeapForJava(), since the restore
+			 * path is only active for releases after Java 8 releases
+			 */
+			candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+		}
+		/* We will set softMx value only if maxHeap calculation returned us a smaller
+		 * value than existing maxHeap or softMx values (or softMx isn't previously set).
+		 * And set the softMx to be the maximum of -Xms (initial memory size) and the
+		 * candidateSoftMx.
```

Comment:
try to have in comments not only what we do, but why.
sometimes it's acutally way more important the 'why', especially if what the code is does is self-explanatory or simple

for example we could  add that  (feel free to simplify/reword)
- we obey explicit -Xmx so to simply give one way to user to override the behavior that uses on softmx
- we don't go bellow initialMemorySize because either heap would otherways be way too small (for default initial values) and be consistent with Xmx logic (when explicitly provided)